src/util/Range.java

```diff
@@ -34,4 +34,66 @@ public int max(int dimension) {
         assert dimension < dimensions();
         return this.max[dimension];
     }
+
+    public boolean contains(int x) {
+        if(numberOfDimensions > 1)
+            return false;
+        if(x < min[0] || x > max[0])
+            return false;
+        return true;
+    }
+
+    public boolean contains(int x, int y) {
+        if (numberOfDimensions < 2 || numberOfDimensions > 2)
+            return false;
+
+            if(x < min[0] || x > max[0] || y < min[1] || y > max[1])
+                return false;
+        return true;
+    }
+
+    @Override
+    public boolean equals(Object arg) {
+        if(!(arg instanceof Range))
+            return false;
+
+        Range other = (Range)arg;
+        if(other.dimensions() != numberOfDimensions) {
+            return false;
+        }
+        else {
+            for(int i = 0; i < numberOfDimensions; i++) {
+                if(other.min[i] != min[i] || other.max[i] != max[i])
+                    return false;
+            }
+            return true;
+        }
+    }
+
+    public boolean includes(Range arg) {
```

Comment:
We like to call same class objects passed as parameter, "that" to contrast with "this" and in this case is more uniform to use the "this" specifier and make the same kind of attribute access (directly or by accessors).